src/main/kotlin/io/snyk/plugin/ui/SnykBalloonNotifications.kt

```diff
@@ -41,15 +47,33 @@ class SnykBalloonNotifications {
                 }
             }
             notification.notify(project)
+            return notification
         }
 
-        fun showSastForOrgEnablement(project: Project) = showInfo(
-            "Snyk Code is disabled by your organisation's configuration. To enable navigate to ",
-            project,
-            NotificationAction.createSimpleExpiring("Snyk > Settings > Snyk Code") {
-                BrowserUtil.browse(getSnykCodeSettingsUrl())
+        fun showSastForOrgEnablement(project: Project): Notification {
+            val notification = showInfo(
+                "Snyk Code is disabled by your organisation's configuration. To enable navigate to ",
+                project,
+                NotificationAction.createSimpleExpiring("Snyk > Settings > Snyk Code") {
+                    BrowserUtil.browse(getSnykCodeSettingsUrl())
+                    startSastEnablementCheckLoop(project)
+                }
+            )
+            var currentAttempt = 1
+            val maxAttempts = 200
+            lateinit var checkIfSastEnabled: () -> Unit
+            checkIfSastEnabled = {
+                if (getApplicationSettingsStateService().sastOnServerEnabled == true) {
```

Comment:
@ArtsiomCh , do we execute here an API call?

Reply:
Just asking settings locally in background, no net call. So presume it's safe.
PS I did check that twice after last week incident :)